File: src/main/java/org/dbshell/environment/EnvironmentVars.java
package org.dbshell.environment;

public class EnvironmentVars {
    public static ContextAndJndi getCurrentContextAndJndi() {
        String envContext = System.getProperty("currentContext");
        String envJndi = System.getProperty("currentJndi");
        return new ContextAndJndi(envContext, envJndi);
    }
    public static void setCurrentContextAndJndi(String context, String jndi) {
        if(context != null)
            System.setProperty("currentContext", context);
        if(jndi != null)
            System.setProperty("currentJndi", jndi);
    }
    public static String getCurrentCatalog() {
        return System.getProperty("currentCatalog");
    }
    public static void setCurrentCatalog(String catalog) {
        if(catalog != null) {
            System.setProperty("currentCatalog", catalog);
        }
    }
}

File: src/main/kotlin/org/dbshell/environment/EnvironmentVars.kt
package org.dbshell.environment

object EnvironmentVars {
    val currentContextAndJndi: ContextAndJndi
        get() {
            val envContext = System.getProperty("currentContext")
            val envJndi = System.getProperty("currentJndi")
            return ContextAndJndi(envContext, envJndi)
        }

    val currentConnectionName: String
        get() {
            val conn = System.getProperty("currentConnectionName")
            return conn
        }

    fun currentConnectionName(name: String?) {
        if(name != null) System.setProperty("currentConnectionName", name)
    }

    fun currentContextAndJndi(context: String?, jndi: String?) {
        if (context != null) System.setProperty("currentContext", context)
        if (jndi != null) System.setProperty("currentJndi", jndi)
    }

    var currentCatalog: String?
        get() = System.getProperty("currentCatalog")
        set(catalog) {
            if (catalog != null) {
                System.setProperty("currentCatalog", catalog)
            }
        }
    var currentSchema: String?
        get() = System.getProperty("currentSchema")
        set(schema) {
            if (schema != null) {
                System.setProperty("currentSchema", schema)
            }
        }
}